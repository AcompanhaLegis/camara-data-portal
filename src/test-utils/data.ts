export const mockedDeputados = (config = {}) => ({
    dados: {
        id: 1,
        ...config,
    }
});
