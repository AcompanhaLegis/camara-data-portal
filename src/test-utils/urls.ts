import { API_URLS, BASE_URL } from "@/api";

export enum URL_COMPLEMENT_NUMBER {
    ID_SUCCESS = 1,
    ID_NOT_FOUND = 2,
    ID_UNKNOWN_ERROR = 3,
}

export enum DEPUTADOS {
    SUCCESS = BASE_URL + API_URLS.DEPUTADOS + "/" + URL_COMPLEMENT_NUMBER.ID_SUCCESS,
    NOT_FOUND = BASE_URL + API_URLS.DEPUTADOS + "/" + URL_COMPLEMENT_NUMBER.ID_NOT_FOUND,
    UNKNOWN_ERROR = BASE_URL + API_URLS.DEPUTADOS + "/" + URL_COMPLEMENT_NUMBER.ID_UNKNOWN_ERROR,
}

export enum PROPOSICOES_TRAMITACOES {
    SUCCESS = BASE_URL + API_URLS.PROPOSICOES + "/" + URL_COMPLEMENT_NUMBER.ID_SUCCESS,
    NOT_FOUND = BASE_URL + API_URLS.PROPOSICOES + "/" + URL_COMPLEMENT_NUMBER.ID_NOT_FOUND,
    UNKNOWN_ERROR = BASE_URL + API_URLS.PROPOSICOES + "/" + URL_COMPLEMENT_NUMBER.ID_UNKNOWN_ERROR,
}