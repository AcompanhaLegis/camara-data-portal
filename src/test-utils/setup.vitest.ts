import { setupServer } from "msw/node";
import { http, HttpResponse } from "msw";
import { beforeAll, afterAll, afterEach } from "vitest";
import { mockedDeputados } from "./data";
import { DEPUTADOS } from "./urls";

// Mocked data
const defaultDeputados = mockedDeputados();

// Mocked handlers
export const restHandlers = [
    http.get(DEPUTADOS.SUCCESS, () => {
        return HttpResponse.json(defaultDeputados);
    }),
    http.get(DEPUTADOS.NOT_FOUND, () => {
        return new HttpResponse(null, { status: 404 });
    }),
    http.get(DEPUTADOS.UNKNOWN_ERROR, () => {
        return new HttpResponse(null, { status: 506 });
    }),
];

// Setup server
const server = setupServer(...restHandlers);

beforeAll(() => server.listen({ onUnhandledRequest: "error" }));
afterEach(() => server.resetHandlers());
afterAll(() => server.close());
