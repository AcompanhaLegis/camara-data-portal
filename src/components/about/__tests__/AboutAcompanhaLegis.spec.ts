import { describe, it } from "vitest";
import { mount } from "@vue/test-utils";

import AboutAcompanhaLegis from "../AboutAcompanhaLegis.vue";


describe("Test AboutAcompanhaLegis component", () => {
    const getWrapper = () => mount(AboutAcompanhaLegis);

    it("should mount", () => {
        const wrapper = getWrapper();
      
        expect(wrapper.vm).toBeDefined();
    });

    it("should match snapshot", () => {
        const wrapper = getWrapper();

        expect(wrapper.html()).toMatchSnapshot();
    });
});

