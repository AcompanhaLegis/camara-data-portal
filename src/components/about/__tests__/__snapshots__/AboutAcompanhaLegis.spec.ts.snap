// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test AboutAcompanhaLegis component > should match snapshot 1`] = `
"<h1 class=\\"mb-6 text-center\\">Sobre o projeto</h1>
<p class=\\"mb-2\\"> O projeto começou como um projeto privado para fornecer dados compilados de proposições e deputados, para que gabinetes pudessem acompanhar as atividades da câmara dos deputados. </p>
<p class=\\"mb-2\\"> Em 2020 projeto foi transformado em um projeto público e aberto (Open Source), para que todos possam acompanhar essas atividades no governo. </p>
<p> Hoje o projeto está em desenvolvimento e utilizamos nosso GitHub para gerenciar o código fonte e o roadmap. </p>"
`;
